armory/datasets/preprocessing.py

```diff
@@ -57,6 +57,40 @@ def carla_over_obj_det_dev(element, modality="rgb"):
     )
 
 
+def carla_video_tracking_preprocess(x, y, y_patch_metadata, max_frames=None):
+    # Clip
+    if max_frames:
+        max_frames = int(max_frames)
+        if max_frames <= 0:
+            raise ValueError(f"max_frames {max_frames} must be > 0")
+        x = x[:max_frames, :]
+        y = y[:max_frames, :]
+        y_patch_metadata = {k: v[:max_frames, :] for (k, v) in y_patch_metadata.items()}
+    # Update labels
+    box_array = tf.squeeze(y, axis=0) if y.shape[0] == 1 else y
+    y = {"boxes": box_array}
+    y_patch_metadata = {
+        k: (tf.squeeze(v, axis=0) if v.shape[0] == 1 else v)
+        for k, v in y_patch_metadata.items()
+    }
+    return x, (y, y_patch_metadata)
+
+
+@register
+def carla_video_tracking_dev(element, max_frames=None):
+    return carla_video_tracking_preprocess(
+        element["video"],
+        element["bboxes"],
+        element["patch_metadata"],
+        max_frames=max_frames,
+    )
+
+
+@register
+def carla_video_tracking_test(element, max_frames=None):
+    return carla_video_tracking_dev(element, max_frames)
```

Comment:
minor refactor suggestion: if `_test`'s preprocessor is just calling `dev`'s, this suggests we can rename `carla_video_tracking_dev` to just `carla_video_tracking`. Then as we do for mnist/resisc/etc:

```
carla_video_tracking_dev = register(carla_video_tracking, "carla_video_tracking_dev")
carla_video_tracking_test = register(carla_video_tracking, "carla_video_tracking_test")
```

and this is a bit simpler by not needing to define functions for both `_dev` and `_test`

Reply:
Addressed in 3ecba7229fafad9140058585b044831932b06122